### DISCUSSION TITLE
What features would you like to see implemented inside the Inbox?

### DISCUSSION THREAD

**Discussion by astubenbord** [astubenbord/paperless-mobile#290] [General]
After setting up a proper paperless workflow, I've come to notice that the inbox in its current state does not really provide a significant advantage compared to simply creating a saved view. 

While the "Swipe to remove from inbox" seemed logical to me when implementing this feature, I often find myself manually removing the tags from the document I just finished editing from within the inbox. 

I am thinking to rework the inbox to make it more useful compared to the gimmick it currently is. Therefore, I would like to know what functionalities you'd like to see in a dedicated inbox. What is _your_ workflow for new documents? 

I'm looking forward to your inputs and ideas!

**Comment by abonhote** [comment]
As I said [here](https://github.com/astubenbord/paperless-mobile/issues/288#issue-1956678018), "swipe to toggle {pick a tag}  tag" would help a lot for the moment.

**Comment by lagset** [comment]
### What's my use case in general?
I often add documents of the same type/source/etc. via paperless-mobile. E.g. account statements out of my mobile banking app or broker app or invoices from my mobile provider app or energy provider.

After adding, paperless (server) does a proper job assigning date, type, tags and so on, but fails one time or another. 

So, I am using the inbox as a "needs review" area and currently adding missing tag, correcting wrongly assigned tags individually for each doc in the inbox.

However, since I am often adding docs from same source/of the same type, it would often be quicker to select all or a subset of docs in the inbox an assign the types/tags etc.

### Mark all read Button
I am currently not using the "mark all read", button, because I review each document and mark it "done" by removing the inbox tag directly after editing, so I have the overview of what's actually "needs review".

### Swipe to remove inbox tag
I like the "swipe to archive" because it fits and is known from other apps, e.g. e-mail. However, I am currently not using the swipe gesture for removing the inbox tag from all or individual docs, because I am editing them in the details view anyway. (This however might change in future, if bulk editing removes the need of editing each individually.)

### further notes
Even though you refer to the current state as gimmicky, I still value the quick access provided by paperless-mobile. A saved view would be far more taps away. 


_____
JFYI: I'll refer to this topic I opened a while ago, dealing with improved bulk editing in the inbox, because it may be relevant for discussion:

https://github.com/astubenbord/paperless-mobile/discussions/222

**Comment by cr0n76** [comment]
I think the most abstracted workflow of the inbox is to check and refine the documents one after another.

In the case everything is okay the existing swipe-to-remove-inbox-tag is already really good. In case all documents are fine, the existing button to mark all as read is really helpful, as well.

But if you need to edit stuff then you need to open the document and first reach a view (with barely more value) where you need to tap on the edit button again. So two UI interactions to achieve what you need -> I think a quick-action button to edit the document would make sense. (And you can think about making the order of the quick-action buttons editable)

Additionally if you edited a document you need to go back two views to see the inbox again. So if you need to edit two documents in a row there are four taps in the app to reach the respective views. -> I think some kind of button / gesture to switch from the document view to the document view of the next document would really help to achieve a way more fluent workflow. In the Paperless-ngx webinterface there is this functionality as well, isn't it?
